app/assets/javascripts/action_plan.js

```diff
@@ -0,0 +1,10 @@
+document.addEventListener("turbolinks:load", function() {
+  $("input[name='action-plan_prefix']").keyup(function(event) {
+    let prefix = event.target.value;
+
+    $("h3.action-plan_heading").each(function(index) {
+      $(this).children("span").remove();
+      $(this).prepend(`<span>${prefix}${prefix.trim() == "" ? "" : index + 1} </span>`);
```

Comment:
you can move the `prefix.trim()` call outside the loop so you don't need to trim the prefix for each iteration